UI/src/model/SelectedMassTransitStop.js

```diff
@@ -277,6 +277,21 @@
         }
     };
 
+    var validateDirectionsForCreation = function () {
+      if(roadCollection) {
+        var nearestLine = geometrycalculator.findNearestLine(roadCollection.getRoadsForMassTransitStops(), currentAsset.payload.lon, currentAsset.payload.lat);
+        var directions_decode = {BothDirections: 1, TowardsDigitizing: 2, AgainstDigitizing: 3};
+        var roadLinkDirection = directions_decode[nearestLine.trafficDirection];
+        var stopDirection = get('validityDirection');
+
+        if(roadLinkDirection != 1)
+          return stopDirection != roadLinkDirection;
+        return true;
+      }else{
+        return false;
+      }
+    };
+
```

Comment:
validateDirectionsForSave and validateDirectionsForCreation have a similar code only with a different return.
Could you probably create a function to return the RoadLinkDirection. Something like that:

var getRoadLinkDirection = function(){
      var nearestLine = geometrycalculator.findNearestLine(roadCollection.getRoadsForMassTransitStops(), currentAsset.payload.lon, currentAsset.payload.lat);
      var linkId = nearestLine.linkId;
      if (!currentAsset.linkId)
        currentAsset.linkId = linkId;
      var directions_decode = {BothDirections: 1, TowardsDigitizing: 2, AgainstDigitizing: 3};
      return directions_decode[nearestLine.trafficDirection];
    };